crates/bevy_core/src/task_pool_options.rs

```diff
@@ -12,6 +13,22 @@ pub struct TaskPoolThreadAssignmentPolicy {
     /// Target using this percentage of total cores, clamped by `min_threads` and `max_threads`. It is
     /// permitted to use 1.0 to try to use all remaining threads
     pub percent: f32,
+    /// Callback that is invoked once for every created thread as it starts
+    #[cfg(not(target_arch = "wasm32"))]
```

Comment:
This would create a compilation failure when retargetting wasm. I think it's probably a better idea to just make this a no-op and documenting it.